internal/expr/parser.go

```diff
@@ -40,6 +56,47 @@ func (ca columnAccessor) String() string {
 	return ca.tableName + "." + ca.columnName
 }
 
+// sliceRangeAccessor represents the expression typeName[low:high].
+type sliceRangeAccessor struct {
+	typ string
+	// Using pointers to represent that the both low and high are optional.
+	low, high *uint64
```

Comment:
These should really be `Optional<uint64>` but Go does not have support for such types. There are several approaches:
1. Store `low` and `high` as numbers and have two flags to indicate whether they are empty or not.
2. Use pointers which are nil if the values are empty.

The advantage of the second approach is that it is enforced by the type system, i.e. you cannot access the number without dereferencing the pointer, thus being reminded to check for null. In the first one, you might forget to check the flag and use the zero value instead.